### DISCUSSION TITLE
Solax G4 discharge to the grid automation

### DISCUSSION THREAD

**Discussion by richwind85** [wills106/homeassistant-solax-modbus#545] [Q&A]
Hi,

I have the Solax modbus integration set up in HA for my G4 Hybrid inverter but Im unsure amount how to go about setting up an automation to discharge the battery to the grid.

I would like to discharge the battery from 9:30pm until 11:30pm or until the battery hits 15%.

Im a complete novice so I really need a guide in how to set this up. 

Thanks in advance

**Comment by wills106** [comment]
I know this link is Gen3 but gives some example automations https://github.com/wills106/homeassistant-solax-modbus/wiki/SolaX-Gen2-&-Gen3-Modes-of-operation#grid-export-gen-3-only

You might want to read about https://github.com/wills106/homeassistant-solax-modbus/wiki/SolaX-Gen4-Modes-of-Operation and also this mode of operation https://github.com/wills106/homeassistant-solax-modbus/wiki/Modbus-Power-Control-(remotecontrol-power)
